### PR TITLE
Added write CSV functionality

### PR DESCRIPTION
**Pull request by iskandari**
- Copied libvol2bird from vol2bird
- Mirrored functionality from vol2bird to write vpts csv if .csv extension found

### CODE REVIEW

**Comment by peterdesmet** [comment]
Nice! Is it possible to drop an example csv output of this function as a comment to this pull request? I can then test if it meets the specification of VPTS CSV.

**Comment by iskandari** [comment]
@peterdesmet here is a sample [output](https://github.com/iskandari/vol2bird/commit/d36008d80c7518f9b9e58e9c29c5b4bb6ac05c17)

**Comment by peterdesmet** [comment]
Thanks, I notice the following things:

1. Move the column `dbz` so columns have the order `gap,eta,dens,dbz,dbz_all`
2. Correct column name typo `radar_wavelenght` to `radar_wavelength`
3. 7 out of the 25 values in `dbz` are much higher than the expected max of 100 (up to 7,385.88)
4. `rcs` and `radar_wavelength` seem to have a false decimal precision of 6 (e.g. `10.710000`). Can this be avoided?
5. `source_file` paths should not be absolute (start with `/`), e.g. `/Users/alexandertedeschi/Documents/CLO/data_pvol/2017/05/04/KHGX/KHGX20170504_013934_V06`. @adokter should we continue to warn for this?
6. 14 out of the 25 values in `dens` are lower than the expected minimum of 0 (up to -29.92)

**Comment by adokter** [comment]
>  `source_file` paths should not be absolute (start with `/`), e.g. `/Users/alexandertedeschi/Documents/CLO/data_pvol/2017/05/04/KHGX/KHGX20170504_013934_V06`. @adokter should we continue to warn for this?

@peterdesmet  - so when source refers to a local file, we use only the base name, and when it refers to an s3 bucket we give the full path? Long paths like this do increase the file size quite a bit, so agree that just the filename when dealing with a locally processed file is better

**Comment by adokter** [comment]
Several of the columns are mixed up, e.g. what is labeled `dbz` is actually `eta`, which is why the validation by @peterdesmet failed

**Comment by peterdesmet** [comment]
> .. what should be the precision required for `rcs`? I was looking at the min constraint `minimum: 1e-15`

That is the minimum because it should not be 0. The precision can be the one defined vol2bird. I've only seen integer values, like `11`.

File: src/libvol2bird/libvol2bird.c [hunk 1]
```diff
@@ -4867,6 +5051,25 @@ int vol2birdLoadConfig(vol2bird_t* alldata, const char* optionsFile) {
 
 #endif
 
+char* get_radar_name(char* source) {
```
Comment (adokter) [inline src/libvol2bird/libvol2bird.c:5054]: change to `int get_radar_name(char* source, char* radarName)`, copying radar name from source into misc.radarName field. Output int can indicate of copy was succesful

**Review by adokter — CHANGES_REQUESTED** [review]

**Comment by peterdesmet** [comment]
Is it possible to post a new `output.csv` here in a comment, so I can validate it?

**Comment by iskandari** [comment]
@peterdesmet since `rcs` can theoretically be a very small number, I don't enforce precision and instead remove trailing zeroes.

> > .. what should be the precision required for `rcs`? I was looking at the min constraint `minimum: 1e-15`
> 
> That is the minimum because it should not be 0. The precision can be the one defined vol2bird. I've only seen integer values, like `11`.

**Pull request merged**